Fix qt_gui_cpp_sip using get_target_property

The build was failing with:

  ld: cannot find -lsiplibqt_gui_cpp_sipcmodule.oTARGET_FILE:tinyxml2::tinyxml2: No such file or directory

The TARGET_FILE CMake generator does not get evaluated in this context.

Upstream-Status: Pending

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

diff --git a/src/qt_gui_cpp_sip/CMakeLists.txt b/src/qt_gui_cpp_sip/CMakeLists.txt
index 89b1e80c..072c85f1 100644
--- a/src/qt_gui_cpp_sip/CMakeLists.txt
+++ b/src/qt_gui_cpp_sip/CMakeLists.txt
@@ -46,8 +46,9 @@ set(_qt_gui_cpp_sip_LIBRARIES
 # sip needs libraries to have resolved paths and cannot link to cmake targets
 foreach(_lib_name ${_qt_gui_cpp_sip_LIBRARIES})
   if(TARGET ${_lib_name})
-    # Use a nifty cmake generator expression to resolve the target location
-    list(APPEND qt_gui_cpp_sip_LIBRARIES $<TARGET_FILE:${_lib_name}>)
+    # Use get_target_property to resolve the target location
+    get_target_property(_lib_location ${_lib_name} LOCATION)
+    list(APPEND qt_gui_cpp_sip_LIBRARIES ${_lib_location})
   else()
     # This library should work as is
     list(APPEND qt_gui_cpp_sip_LIBRARIES ${_lib_name})
